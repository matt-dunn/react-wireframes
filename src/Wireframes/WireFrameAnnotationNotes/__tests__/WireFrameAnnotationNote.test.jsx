/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import React from "react";
import { shallow } from "enzyme";

import { WireFrameAnnotationsNote as Component } from "../WireFrameAnnotationNote";

describe("Wireframe: WireFrameAnnotationsNote", () => {
  let MockedComponent;
  let component;

  beforeEach(() => {
    MockedComponent = jest.fn();

    component = {
      id: 1,
      count: 1,
      Component: MockedComponent,
      options: {
        title: "Test component",
        description: "Test description",
      },
    };
  });

  it("should render correctly", () => {
    const wrapper = shallow(
      <Component
        component={component}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
