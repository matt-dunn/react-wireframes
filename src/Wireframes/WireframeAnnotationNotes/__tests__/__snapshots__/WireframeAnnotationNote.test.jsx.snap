// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wireframe: WireframeAnnotationsNote should render correctly 1`] = `
<WireframeAnnotationsNoteContainer
  isHighlighted={false}
>
  <header>
    <IdentifierNote
      annotation={
        Object {
          "Component": [MockFunction],
          "count": 1,
          "id": 1,
          "options": Object {
            "description": "Test description",
            "title": "Test component",
          },
        }
      }
    />
    <h2>
      Test component
    </h2>
  </header>
  <article>
    Test description
  </article>
</WireframeAnnotationsNoteContainer>
`;
